File: src/main/java/pe/edu/vallegrande/arquitectura/model/Product.java
package pe.edu.vallegrande.arquitectura.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@Data
@NoArgsConstructor
@Table(name="product")
public class Product {
    @Id
    @Column(name = "id_product")
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long idProduct;

    @Column(name = "\"name\"")
    private String name;

    @Lob
    @Column(name = "\"specification\"")
    private String specification;

    @Lob
    @Column(name = "\"description\"")
    private String description;

    @Column(name = "\"category\"")
    private String category;

    @Column(name = "unit_price")
    private Double unitPrice;

    @Column(name = "stock")
    private int stock;

    @Column(name = "\"status\"")
    private String status;

    @Column(name = "internal_code")
    private String internalCode;

    @Column(name = "registration_date")
    private LocalDateTime registrationDate;

    @Column(name = "brand_id")
    private Long brandId; // o private int brandId; si es NOT NULL en DB
}

File: src/app/core/interfaces/product.ts
export interface Product {
    idProduct?: number;         // 'Long' en Java -> 'number' en TypeScript
    name: string;
    specification: string;
    description: string;
    category: string;
    unitPrice: number;          // 'Double' en Java -> 'number' en TypeScript
    stock: number;              // ¡CORREGIDO! 'int/Integer' en Java -> 'number' en TypeScript
    status: string;
    internalCode: string;
    registrationDate: string;   // 'LocalDateTime' en Java -> 'string' (ISO 8601) en TypeScript
    brandId: number;            // ¡CORREGIDO! 'int/Long' en Java -> 'number' en TypeScript
}
